MODULE.bazel

```diff
@@ -0,0 +1,332 @@
+module(name = "buildbuddy")
+
+bazel_dep(name = "bazel_skylib", version = "1.5.0")
+bazel_dep(name = "platforms", version = "0.0.8")
+bazel_dep(name = "rules_proto", version = "5.3.0-21.7")
+
+# Needed for com_google_protobuf
+bazel_dep(name = "rules_python", version = "0.31.0")
+
+python = use_extension("@rules_python//python/extensions:python.bzl", "python")
+python.toolchain(
+    configure_coverage_tool = False,
+    # Do not error if root user is detected
+    ignore_root_user_error = True,
+    python_version = "3.11",
+)
+
+bazel_dep(name = "zlib", version = "1.3")
+bazel_dep(name = "bazel_features", version = "1.4.1", repo_name = "io_bazel_rules_go_bazel_features")
+
+version = use_extension("@io_bazel_rules_go_bazel_features//private:extensions.bzl", "version_extension")
+use_repo(version, "bazel_features_globals", "bazel_features_version")
+
+bazel_dep(name = "rules_go", version = "0.46.0", repo_name = "io_bazel_rules_go")
+bazel_dep(name = "gazelle", version = "0.35.0", repo_name = "bazel_gazelle")
+archive_override(
+    module_name = "gazelle",
+    integrity = "sha256-MpOL2hbmcABjA1R5Bj2dJMYO2o15/Uc5Vj9Q0zHLMgk=",
+    patch_strip = 1,
+    patches = [
+        "@@//buildpatches:gazelle.patch",
+    ],
+    urls = [
+        "https://mirror.bazel.build/github.com/bazelbuild/bazel-gazelle/releases/download/v0.35.0/bazel-gazelle-v0.35.0.tar.gz",
+        "https://github.com/bazelbuild/bazel-gazelle/releases/download/v0.35.0/bazel-gazelle-v0.35.0.tar.gz",
+    ],
+)
+
+go_sdk = use_extension("@io_bazel_rules_go//go:extensions.bzl", "go_sdk")
+go_sdk.download(
+    goarch = "arm64",
+    goos = "linux",
+    version = "1.22.1",
+)
+go_sdk.download(
+    goarch = "amd64",
+    goos = "linux",
+    version = "1.22.1",
+)
+go_sdk.download(
+    goarch = "arm64",
+    goos = "darwin",
+    version = "1.22.1",
+)
+go_sdk.download(
+    goarch = "amd64",
+    goos = "darwin",
+    version = "1.22.1",
+)
+go_sdk.nogo(nogo = "@//:vet")
+use_repo(
+    go_sdk,
+    "go_toolchains",
+    "io_bazel_rules_nogo",
+)
+
+go_deps = use_extension("@bazel_gazelle//:extensions.bzl", "go_deps")
+go_deps.from_file(go_mod = "//:go.mod")
+
+# Go repos with patches
+go_deps.module_override(
+    patch_strip = 1,
+    patches = [
+        "@io_bazel_rules_go//third_party:com_github_gogo_protobuf-gazelle.patch",
+    ],
+    path = "github.com/gogo/protobuf",
+)
+go_deps.gazelle_override(
+    build_file_generation = "off",
+    path = "github.com/gogo/protobuf",
+)
+go_deps.module_override(
```

Comment:
You can get `circl` via `bazel_dep(name = "circl")`, which has the patches to get the CGO build to work. If these patches here do anything more than that, we can of course keep it.

Reply:
I will defer this to a later PR.

We are building with both bzlmod turned on and off right now so I don't want to have inconsistency between the 2 modes.